### PR TITLE
Add @metamask/test-dapp

### PR DESCRIPTION
**Pull request by rekmarks**
- Add `@metamask/test-dapp` for use in e2e tests
- Delete `e2e/contract-test` directory, which held the local copy of the test dapp

### CODE REVIEW

**Comment by whymarrh** [comment]
Rebasing this on the latest `develop` should get the build going

**Comment by rekmarks** [comment]
Thanks for the reminder @whymarrh, it works!

**Review by whymarrh — DISMISSED** [review]
LGTM!

**Comment by Gudahtt** [comment]
Is the onboarding library still being loaded from `node_modules` when this is run locally? 🤔 That's the one thing I'm not sure on, since it's not currently tested and the test dapp itself is tolerant of that being broken.

**Comment by rekmarks** [comment]
@Gudahtt it is, per:

https://github.com/MetaMask/test-dapp/blob/master/website/index.html#L231

The `..node_modules` path is replaced with a CDN link when we deploy the test dapp to GitHub Pages.

File: yarn.lock [hunk 1]
```diff
@@ -1910,6 +1910,20 @@
   dependencies:
     bowser "^2.5.4"
 
+"@metamask/onboarding@^0.2.1":
+  version "0.2.1"
```
Comment (Gudahtt) [inline yarn.lock:1914]: Oh, could to dedupe the above version of this as well, and remove it from the manifest? It's no longer used anywhere.

**Review by Gudahtt — CHANGES_REQUESTED** [review]

**Comment by Gudahtt** [comment]
👍 I checked it out locally, it's all working fine except that it's loading the wrong version of the onboarding library, which deduping it would solve

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by whymarrh — APPROVED** [review]
LGTM!

**Pull request merged**